### PR TITLE
Wait to raise on mismatched filesizes

### PR DESCRIPTION
**Pull request by jasoncpatton**
Instead of raising `InconsistentManifest` immediately if there is a
filesize mismatch between a local file and the metadata in the
transfer_manifest, we now store the filepath in a dict, and delete
the key if a later entry does not lead to a mismatch. Then raise
if there are any entries left in the dict at the end of the analysis.

This means that only the last filesize check will matter, so if a
new file is transferred with the same filepath, we don't immediately
throw an exception and stop analyzing the manifest.

### CODE REVIEW

File: xfer.py [hunk 1]
```diff
@@ -994,11 +995,10 @@ def analyze(transfer_manifest_path: Path) -> None:
...
             if local_size != size:
-                raise InconsistentManifest(
-                    "Local size of {} of {} does not match anticipated size {}.".format(
-                        local_size, fname, size,
-                    )
-                )
+                mismatched_filesizes[fname] = {"expected": size, "got": local_size}
+            else:
+                if fname in mismatched_filesizes:
+                    del mismatched_filesizes[fname]
```
Comment (JoshKarpel) [inline xfer.py]:
```suggestion
                mismatched_filesizes.pop(fname, None)
```
is a little cleaner IMO

File: xfer.py [hunk 2]
```diff
@@ -1032,6 +1032,18 @@ def analyze(transfer_manifest_path: Path) -> None:
...
             sync_request = {"files": {}, "transfer_files": set(), "verified_files": {}}
 
+    if len(mismatched_filesizes) > 0:
+        for fname, size in mismatched_filesizes.items():
+            logging.error(
+                "- Mismatched file size for %s: expected %d, got %d on disk",
+                fname,
+                size["expected"],
+                size["got"],
+            )
```
Comment (JoshKarpel) [inline xfer.py]: Maybe capture the line number of the mismatch as well for debugging purposes?

**Review by JoshKarpel — CHANGES_REQUESTED** [review]
A few tweaks...

File: xfer.py (hunk 2 above)
Comment (jasoncpatton) [reply xfer.py]: Marked this as resolved early, oops, but this is added now

**Review by JoshKarpel — APPROVED** [review]
LGTM!

**Pull request merged**